.grit/patterns/go/channel_guarded_with_mutex.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Remove the mutex. Channel guarded with mutex
+---
+
+Detected a channel guarded with a `mutex`. Channels already have an internal `mutex`, so this is unnecessary. Remove the mutex
+
+### references
+
+- [go-antipatterns](https://hackmysql.com/golang/go-antipatterns/#guarded-channel)
+
+tags: #fix #best-practice
+
+```grit
+language go
+
+`` as $body where {
```

Comment:
This isn't really a good function, as `` is a very expensive query to match - it matches *everything*, including recursively.

Statement blocks probably work here. `{ $body }`

Reply:
Hi @morgante, yes I was also thinking same, `` will parse the whole file which will become expansive, updated it now.